### PR TITLE
Add py310 testing to CI

### LINKED ISSUE #160: Support for py36?
It's currently unclear whether py36 is really supported or not. The setup configuration says so, but the testing does not cover it.

Py36 dies in 3 months, so maybe it's best we just support py37+. See: https://endoflife.date/python

### PR DESCRIPTION
**Pull request by cthoyt**
Closes #160

Testing makes it clear that the dependencies of this package do not run on py36, so it's kicked out.

Py310 support blocked by:
- https://github.com/linkml/linkml/pull/401
- https://github.com/linkml/linkml-model/issues/49
- and probably more, since linkml is split among so many repos and they're _really_ hard to understand

### CODE REVIEW

**Comment by matentzn** [comment]
@cthoyt we are getting ready to merge the remaining PRs.. do you want to try and finalise this one?

**Comment by cthoyt** [comment]
I believe this PR is blocked by upstream repositories not yet having Python 3.10 support. CI is running now and will report if that's still the cas

File: .github/workflows/qc.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
```
Comment (matentzn) [inline .github/workflows/qc.yml:16]: Why is 3.10 in quotes?

**Comment by matentzn** [comment]
Thank you @cthoyt!

File: .github/workflows/qc.yml (hunk 1 above)
Comment (cthoyt) [reply .github/workflows/qc.yml:16]: 3.10 not in quotes is the same as 3.1

Comment (matentzn) [reply .github/workflows/qc.yml:16]: Very interesting. Thanks

File: .github/workflows/qc.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: ["3.7", "3.10"]
```
Comment (matentzn) [inline .github/workflows/qc.yml:16]: Just for me to understand, why did you remove 3.8, 3.9?

Comment (cthoyt) [reply .github/workflows/qc.yml:16]: If it works on 3.7 and 3.10 it's pretty good chance it works in the middle. I also needed to make a change to get it to run again 😆 . We're still not ready for 3.10 apparently

**Comment by hrshdhgd** [comment]
Solved by #261
